aten/src/ATen/native/ForeachOpsKernels.cpp

```diff
@@ -232,4 +235,26 @@ std::vector<Tensor> foreach_tensor_norm_slow(TensorList tensors, const Scalar& o
   return result;
 }
 
+Tensor foreach_tensor_global_norm_slow(TensorList tensors, const Scalar& ord) {
+  TORCH_CHECK((ord.isIntegral(false) || ord.isFloatingPoint()), "foreach_norm supports int and float ord");
+  double p;
+  if (ord.isIntegral(false)) {
+    p = static_cast<int64_t>(ord.to<int64_t>());
+  }
+  if (ord.isFloatingPoint()) {
+    p = ord.to<double>();
+  }
+  check_foreach_api_restrictions(tensors);
+  const auto result_scalar_type = toRealValueType(result_type(tensors));
+  const auto result = std::accumulate(
+    tensors.begin(), tensors.end(),
+    at::zeros({}, tensors[0].options().dtype(toOpMathType(result_scalar_type))),
+    [&](const Tensor &cumulative_sum, const Tensor &t) {
+      auto norm = at::linalg_vector_norm(t, ord);
+      return cumulative_sum + (p == 0.0 || p == 1.0 ? norm : at::pow(norm, ord));
+    }
+  );
+  return (p == 0.0 ? result : at::pow(result, 1 / p)).to(result_scalar_type);
```

Comment:
do you need `pow` if p=1?

Reply:
it doesn't sound needed and the test cases look okay